### PR TITLE
$VERBOSE should admit truthy values.

### PR DESCRIPTION
**Pull request by nirvdrum**

### CODE REVIEW

File: src/main/java/org/truffleruby/language/globals/UpdateVerbosityNode.java [hunk 1]
```diff
@@ -9,31 +9,31 @@
...
 
-    @TruffleBoundary
-    private void validateVerbose(Object value) {
-        if (value instanceof Boolean || value == nil()) {
-            // OK
+        if (nilProfile.profile(value == nil())) {
+            return value;
         } else {
-            throw new UnsupportedOperationException();
+            return booleanCastNode.executeToBoolean(value);
```
Comment (eregon) [inline src/main/java/org/truffleruby/language/globals/UpdateVerbosityNode.java]: I was wondering if MRI casts as well and it seems it does:
```ruby
[1] pry(main)> $VERBOSE
=> false
[2] pry(main)> $VERBOSE = 42
=> 42
[3] pry(main)> $VERBOSE 
=> true
```

Could you maybe add a spec for this?

Comment (nirvdrum) [reply src/main/java/org/truffleruby/language/globals/UpdateVerbosityNode.java]: I was a bit surprised to see there really wasn't anything testing `$VERBOSE`. Where would you want it? predefined_spec.rb?

Comment (eregon) [reply src/main/java/org/truffleruby/language/globals/UpdateVerbosityNode.java]: Yes.

Comment (nirvdrum) [reply src/main/java/org/truffleruby/language/globals/UpdateVerbosityNode.java]: Spec added. I'll let you take a look at that before I merge.

File: spec/ruby/language/predefined_spec.rb [hunk 2]
```diff
@@ -879,6 +879,23 @@ def obj.foo2; yield; end
   end
 end
 
+describe "Global variable $VERBOSE" do
+  it "mostly allows truthy values" do
```
Comment (eregon) [inline spec/ruby/language/predefined_spec.rb]: Why `mostly`?

Comment (eregon) [reply spec/ruby/language/predefined_spec.rb]: I think something like `converts truthy values to true` would be more descriptive of the behavior (and the false case would need to be split then).

File: src/main/java/org/truffleruby/language/globals/UpdateVerbosityNode.java (hunk 1 above)
Comment (eregon) [reply src/main/java/org/truffleruby/language/globals/UpdateVerbosityNode.java]: Thanks!

File: spec/ruby/language/predefined_spec.rb (hunk 2 above)
Comment (nirvdrum) [reply spec/ruby/language/predefined_spec.rb]: I think the name "truthy" has come to mean handling both `true` and `false` cases, which is why I said mostly. But your suggestion sounds clearer.

**Pull request merged**